contrib/ray/raycluster_example.yaml

```diff
@@ -1,60 +1,112 @@
+    port: 6383
+    appProtocol: grpc
+  - name: dashboard-agent-listen-port
+    port: 52365
+    appProtocol: http
+  - name: metrics-export-port
+    port: 8080
+    appProtocol: http
+  - name: p10002
+    port: 10002
+    appProtocol: grpc
+  - name: p10003
+    port: 10003
+    appProtocol: grpc
+  - name: p10004
+    port: 10004
+    appProtocol: grpc
+  - name: p10005
+    port: 10005
+    appProtocol: grpc
+  - name: p10006
+    port: 10006
+    appProtocol: grpc
+  - name: p10007
+    port: 10007
+    appProtocol: grpc
+  - name: p10008
+    port: 10008
+    appProtocol: grpc
+  - name: p10009
+    port: 10009
+    appProtocol: grpc
+  - name: p10010
+    port: 10010
+    appProtocol: grpc
+  - name: p10011
+    port: 10011
+    appProtocol: grpc
+  - name: p10012
+    port: 10012
+    appProtocol: grpc
+---
 apiVersion: ray.io/v1
 kind: RayCluster
 metadata:
   name: kubeflow-raycluster
 spec:
   rayVersion: '2.23.0'
-  # If `enableInTreeAutoscaling` is true, the Autoscaler sidecar will be added to the Ray head pod.
   enableInTreeAutoscaling: true
-  # `autoscalerOptions` is an OPTIONAL field specifying configuration overrides for the Ray Autoscaler.
-  # The example configuration shown below below represents the DEFAULT values.
-  # (You may delete autoscalerOptions if the defaults are suitable.)
   autoscalerOptions:
-    # Default: Upscaling is not rate-limited. This mode adds new worker pods to handle increased workload as quiclky as possible.
     upscalingMode: Default 
-    # `idleTimeoutSeconds` is the number of seconds to wait before scaling down a worker pod which is not using Ray resources.
     idleTimeoutSeconds: 60
-  # Ray head pod configuration
   headGroupSpec:
-    # Kubernetes Service Type.
-    serviceType: ClusterIP
-    # The following params are used to complete the ray start: ray start --head --block --dashboard-host: '0.0.0.0' ...
     rayStartParams:
-      # Setting "num-cpus: 0" to avoid any Ray actors or tasks being scheduled on the Ray head Pod.
-      num-cpus: "0"
-      dashboard-host: '0.0.0.0'
-      block: 'true'
-    # pod template
+      num-cpus: '1'
+      node-manager-port: '6380'
+      object-manager-port: '6381'
+      runtime-env-agent-port: '6382'
+      dashboard-agent-grpc-port: '6383'
+      dashboard-agent-listen-port: '52365'
+      metrics-export-port: '8080'
+      max-worker-port: '10012'
+      node-ip-address: $(hostname -I | tr -d ' ' | sed 's/\./-/g').raycluster-istio-headless-svc.kubeflow-user-example-com.svc.cluster.local
     template:
-      metadata:
-        # Custom labels. NOTE: To avoid conflicts with KubeRay operator, do not define custom labels start with `raycluster`.
-        # Refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
-        # The ray head must not have an Istio sidecar
-        # TODO add an authorizationpolicy in the future for the ray head
-        labels:
-          sidecar.istio.io/inject: "false"
```

Comment:
can you keep 

```
        labels:
          sidecar.istio.io/inject: "true"
```
to have it explicit ?

Reply:
Did it